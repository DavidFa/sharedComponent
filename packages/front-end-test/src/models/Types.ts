export interface PostType {
    id: number;
    title: string;
    body: string
}

export enum PostStatus {
    list,
    edit,
    add
}