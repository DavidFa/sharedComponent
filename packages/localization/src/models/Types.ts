export type InputFormModel = {
    name: InputFormProps;
    description: InputFormProps;
}


export type InputFormProps = {
    label: string;
    placeholder: string;
    length: number;
}

export type InputFormWithLanguageProps = {
    language: string;
    placeholder: string;
    length: number;
}

export interface LocalizationState {
    language: string;
    languages: string[];
}

export enum LocalizationActionType {
    updateLanguage,
    populateLanguages
}

export type LocalizationAction = {
    type: LocalizationActionType;
    payload: {
        language?: string;
        languages?: string[];
    }
}