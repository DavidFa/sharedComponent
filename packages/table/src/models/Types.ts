export type DataGridColumns = {
    field: string;
    headerName: string;
    width: number;
    align?: string;
    sortable?: boolean;
}

export type DataGridRows = {
    [key: string]: string | number
}