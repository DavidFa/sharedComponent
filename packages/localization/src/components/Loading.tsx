import React from "react";
import styled from "styled-components";
import { useAppSelector } from "../hooks/hooks";
import classes from "./Loading.module.css";

const Wrapper = styled.div`
text-align:center;
`;


const Loading: React.FC = () => {
    const loading = useAppSelector(state => state.message.loading);
    return <Wrapper>{loading && <div className={classes.loading}>Loading...</div>}</Wrapper>;
}

export default Loading;