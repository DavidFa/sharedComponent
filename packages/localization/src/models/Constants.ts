export const FIELD_NAME: string = "Name";
export const FIELD_DESC: string = "Description";