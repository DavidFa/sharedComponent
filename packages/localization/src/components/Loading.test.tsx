import Loading from "./Loading";
import configureStore from 'redux-mock-store';
import { Status } from '../models/Types';
import { AnyAction, Store } from "redux";
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

const mockStore = configureStore([]);

describe('Loading Component', () => {
    it('should render with given state from Redux State', () => {
        const store: Store<any, AnyAction> = mockStore({
            message: {
                loading: true,
                status: Status.init,
                message: "",
            }
        })

        const component = renderer.create(
            <Provider store={store}>
                <Loading />
            </Provider>
        )

        expect(component.toJSON()).toMatchSnapshot();
    })

    it('should render without Loading', () => {
        const store = mockStore({
            message: {
                loading: false,
                status: Status.init,
                message: "",
            }
        })

        const component = renderer.create(
            <Provider store={store}>
                <Loading />
            </Provider>
        )

        expect(component.toJSON()).toMatchSnapshot();
    })


    it('should contain a loading message', () => {
        const store = mockStore({
            message: {
                loading: true,
                message: "",
                status: Status.init
            }
        })

        render(<Provider store={store}>
            <Loading />
        </Provider>)

        const div = screen.queryByText("Loading", { exact: false });
        expect(div).toBeInTheDocument();
    })

    it('should not contain a loading message', () => {
        const store = mockStore({
            message: {
                loading: false,
                message: "",
                status: Status.init
            }
        })

        render(<Provider store={store}>
            <Loading />
        </Provider>)

        const div = screen.queryByText("Loading", { exact: false });
        expect(div).not.toBeInTheDocument();
    })
})

