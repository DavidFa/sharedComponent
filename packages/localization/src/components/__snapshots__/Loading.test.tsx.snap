// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component should render with given state from Redux State 1`] = `
<div
  className="sc-bdfBQB gvSErf"
>
  <div
    className="loading"
  >
    Loading...
  </div>
</div>
`;

exports[`Loading Component should render without Loading 1`] = `
<div
  className="sc-bdfBQB gvSErf"
/>
`;
